app/policies/request_policy.rb

```diff
@@ -4,26 +4,21 @@ class Scope < ApplicationPolicy::Scope
     PERSONA_APPROVER  = 'approval/approver'.freeze
     PERSONA_REQUESTER = 'approval/requester'.freeze
 
-    def resolve
-      return scope.all unless user.rbac_enabled?
-
+    def resolve_scope
       if user.params[:request_id]
         req = Request.find(user.params[:request_id])
         raise Exceptions::NotAuthorizedError, "Read access not authorized for request #{req.id}" unless resource_check('read', req)
-        req.children
-      # GraphQL comes in with model_class.all, whose scope is respond to model call
-      # Regular root requests call comes in with model class itself
+        req.requests
       else
         case Insights::API::Common::Request.current.headers[Insights::API::Common::Request::PERSONA_KEY]
         when PERSONA_ADMIN
           raise Exceptions::NotAuthorizedError, "No permission to access the complete list of requests" unless admin?(scope)
-          scope.respond_to?(:model) ? scope : scope.where(:parent_id => nil)
+          scope.where(:parent_id => nil)
```

Comment:
@hsong-rh Does it make sense to add a scope in the app/model/requests for top_level_requests so we dont have to have parent_id => null logic in multiple places.

Similar to

https://github.com/RedHatInsights/approval-api/blob/5fe66adc87f6df620bbdec90ba750beb3dd8025e/app/models/concerns/owner_field.rb#L8